content/learn/08.contributions/01.arduino-library-style-guide/arduino-library-style-guide.md

```diff
@@ -0,0 +1,74 @@
+
+**Avoid words that have different meanings to the general public.** For example, to programmers, an error is a notification that something happened. To the general public, errors are bad things.
+
+**When you have to use a domain-specific term, write a sentence or two describing it to the general public FIRST.** You’ll likely come across a better term, and if not, you’ll have started the documentation on your library.
+
+**Document and comment as you go.** When writing examples and documentation, follow the [Writing Style Guide](/content/learn/07.creating-content/ArduinoWritingStyleGuide/ArduinoWritingStyleGuide.md)
+
+**Use the established core libraries and styles.**
+
+* Use `read()` to read inputs, and `write()` to write to outputs, e.g. `digitalRead()`, `analogWrite()`, etc.
+* Use the `Stream.h` and `Print.h` libraries when dealing with byte streams. If it’s not appropriate, at least try to use its API as a model. For more on this, see below
+* For network applications, use the Client and Server libraries as the basis.
+* Use `begin()` to initialize a library instance, usually with some settings. Use `end()` to stop it.
+* Use camel case function names, not underscore. For example, **analogRead**, not **analog_read**. Or **myNewFunction**, not **my_new_function**. We've adopted this from Processing.org for readability's sake.
+
+**LONG_CONSTANT_NAMES_FULL_OF_CAPS are hard to read.** Try to simplify when possible, without being terse.
+
+**Try to avoid boolean arguments.** Instead, consider providing two different functions with names the describe the differences between them.
+
+**Don’t assume knowledge of pointers.** Beginning users of C find this the biggest roadblock, and get very confused by & and *, so whenever you can avoid having them hanging out in the API, do so. One way is to pass by reference using array notation rather than * notation, for example.
+
+```arduino
+   void printArray( char* array);
+```
+
+can be replaced by
+
+```arduino
+    void printArray(char[] array);
+```
+
+Though there are some libraries where we pass pointers by using structures like const chars, avoid anything that requires the user to pass them. For example,rather than:
+
+```arduino
+  foo.readAccel(&x, &y, &z);
+```
+
+use something like this:
+
+```arduino
+   xAxis = adxl.readX();
+   yAxis = adxl.readY();
+   zAxis = adxl.readZ();

```

Comment:
@karlsoderby do we know why this happens?

Reply:
@jhansson-ard there is just this one remaining suggestion.

If the problem with the rendering is to be fixed separately, that is fine. I only noted it here because I noticed it in the preview. But I would like to get the style issue of the non-standard three space indents resolved.

Reply:
Hmm I do not know why this happens. I will report it to the WEDO team so that they can investigate when they have the time.

Reply:
@per1234 we will report this to another team, as it is a bigger bug than we can take care of. Thanks for finding it though!